### ISSUE TITLE
drive: authError when uploading big files

### ISSUE THREAD

**Issue by ncw** [ncw/rclone#33]
Factored from #28 

Eg from @benfry

```
(50GB file)
2015/02/27 17:17:22 
Transferred: 51805750272 Bytes (12582.34 kByte/s)
Errors: 0
Checks: 0
Transferred: 0
Elapsed time: 1h7m0.837699975s
Transferring: Blade2_C_Drive006.v2i

2015/02/27 17:17:54 Blade2_C_Drive006.v2i: Failed to copy: Upload failed: googleapi: Error 401: Invalid Credentials, authError

Transferred: 52428762515 Bytes (12631.21 kByte/s)
Errors: 1
Checks: 0
Transferred: 1
Elapsed time: 1h7m33.449969925s
```

This can be reproduced very easily using the `--bwlimit` flag to make transfer take an arbitrarily long file.

The auth errors look very much like these problems

  * http://stackoverflow.com/questions/23789284/resumable-upload-error-401
  * https://code.google.com/p/google-api-python-client/issues/detail?id=231

I have tried
  * refreshing the auth every 5 minutes during long transfers
    * doesn't work - as it doesn't refresh the auth that is in progress
  * [using the Resumable Media API](http://godoc.org/github.com/google/google-api-go-client/drive/v2#FilesInsertCall.ResumableMedia) in the google drive API
    * doesn't work as it isn't finished according to the author

That leaves using the chunked API
  * https://developers.google.com/drive/web/manage-uploads#resumable

NB might need to do this for google cloud storage too which probably has the same underlying problem

**Comment by madralphw** [comment]
New attempt:

# for i in `seq 1 30`; do rclone  --stats=40m0s  --transfers=1 copy Blade2_C_Drive006.v2i drive:/blade2/ ; done 
2015/02/27 20:13:53 Google drive root 'blade2': Building file list
2015/02/27 20:13:54 Google drive root 'blade2': Waiting for checks to finish
2015/02/27 20:13:54 Google drive root 'blade2': Waiting for transfers to finish
2015/02/27 20:53:53 
Transferred:   40833057792 Bytes (16608.94 kByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:  40m0.877676827s
Transferring:  Blade2_C_Drive006.v2i

2015/02/27 21:02:22 Blade2_C_Drive006.v2i: Failed to copy: Upload failed: googleapi: Error 401: Invalid Credentials, authError

Transferred:   52428762515 Bytes (17594.07 kByte/s)
Errors:                 1
Checks:                 0
Transferred:            1
Elapsed time:  48m30.068723257s

2015/02/27 21:02:23 Google drive root 'blade2': Building file list
2015/02/27 21:02:24 Google drive root 'blade2': Waiting for checks to finish
2015/02/27 21:02:24 Google drive root 'blade2': Waiting for transfers to finish
2015/02/27 21:42:23 
Transferred:   50281972736 Bytes (20453.44 kByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:  40m0.744756276s
Transferring:  Blade2_C_Drive006.v2i


Transferred:   52428762515 Bytes (20432.36 kByte/s)
Errors:                 0
Checks:                 0
Transferred:            1
Elapsed time:  41m45.827492424s

2015/02/27 21:44:09 Google drive root 'blade2': Building file list
2015/02/27 21:44:10 Google drive root 'blade2': Waiting for checks to finish
2015/02/27 21:44:10 Google drive root 'blade2': Waiting for transfers to finish

Transferred:            0 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:                 1
Transferred:            0
Elapsed time:  1.402673446s

The second attempt worked ;-)

**Comment by ncw** [comment]
@madralphw I'm currently testing a fix for this which is looking encouraging.  If you want to try a test binary, then drop me an email to nick@craig-wood.com  with your platform (windows/linux 32/64 bit) and I'll send you one.

**Comment by ncw** [comment]
I've fixed this in rcone v1.12 which you can download from rclone.org

It switches to chunked mode for uploading big files which should make big file uploading much more reliable, and mean you won't get the auth problems any more.

Please re-open this ticket if you see the auth problem again.

Thanks for the report

Nick

**Comment by madralphw** [comment]
Thanks ;-)  I'll try it now.
